pkg/models/devops/s2ibinary_handler.go

```diff
@@ -56,12 +57,11 @@ type s2iBinaryUploader struct {
 	s3Client  s3.Interface
 }
 
-func NewS2iBinaryUploader(client versioned.Interface, informers externalversions.SharedInformerFactory, s3Client s3.Interface,
+func NewS2iBinaryUploader(client versioned.Interface, s3Client s3.Interface,
 	k8sClient k8s.Client) S2iBinaryUploader {
 	return &s2iBinaryUploader{
 		k8sClient: k8sClient,
 		client:    client,
-		informers: informers,
```

Comment:
You deleted the variable `informers`. But the struct `s2iBinaryUploader` still has it. See also https://github.com/kubesphere-sigs/ks-devops/blob/1102edae144c1dbe4c2005059be9e3b7daaa2bb2/pkg/models/devops/s2ibinary_handler.go#L55

So, it will cause nil exceptions. See the following screenshot:

![image](https://user-images.githubusercontent.com/1450685/126744591-9c3864ad-4466-46b2-9df2-4f5666cf690a.png)

In order to make it works well. You need to provide an alternative way to get k8s resources. Such as  https://github.com/kubesphere-sigs/ks-devops/pull/32/files